lib/HLSL/DxilContainerReflection.cpp

```diff
@@ -1097,7 +1097,7 @@ HRESULT CShaderReflectionType::Initialize(
 
       // There is no annotation for empty structs
       unsigned int fieldCount = 0;
-      if (structAnnotation)
+      if (structAnnotation && !structAnnotation->IsEmptyBesidesResources())
```

Comment:
Otherwise we try to drill into empty struct fields, which have no type annotation, and then we fail.  There's no point to drill into a struct with no numeric elements in reflection.